channeld/full_channel.c

```diff
@@ -353,11 +353,11 @@ struct bitcoin_tx **channel_splice_txs(const tal_t *ctx,
 	other_side_pay = channel->view[side].owed[!side];
 
 	if (side == LOCAL) {
-		side_pay.millisatoshis += splice_amnt * 1000;
-		other_side_pay.millisatoshis += remote_splice_amnt * 1000;
+		side_pay.millisatoshis += splice_amnt * 1000; /* Raw: splicing */
+		other_side_pay.millisatoshis += remote_splice_amnt * 1000; /* Raw: splicing */
```

Comment:
The reason we have amount_sat_to_msat() is to make sure you handle overflows.  You should think about what to do in that case!!

Reply:
Changed it to this style:

amount.h
```
WARN_UNUSED_RESULT bool amount_msat_add_sat_s64(struct amount_msat *val,
						struct amount_msat a,
						struct s64 b);
```

amount.c
```
WARN_UNUSED_RESULT bool amount_msat_add_sat_s64(struct amount_msat *val,
						struct amount_msat a,
						struct s64 b)
{
	if (b < 0)
		return amount_msat_sub_sat(val, a, amount_sat(-b));
	else
		return amount_msat_add_sat(val, a, amount_sat(b));
}
```

full_channel.c
```
	if (side == LOCAL) {
		if (!amount_msat_add_sat_s64(&side_pay, side_pay, splice_amnt))
			return NULL;
		if (!amount_msat_add_sat_s64(&other_side_pay, other_side_pay, remote_splice_amnt))
			return NULL;
	} else if (side == REMOTE) {
		if (!amount_msat_add_sat_s64(&side_pay, side_pay, remote_splice_amnt))
			return NULL;
		if (!amount_msat_add_sat_s64(&other_side_pay, other_side_pay, splice_amnt))
			return NULL;
	}
```